apps/events/models.py

```diff
@@ -78,13 +103,19 @@ def trigger_type(self):
 
     def fire(self, session):
         try:
-            result = ACTION_FUNCTIONS[self.action.action_type](session, self.action.params)
+            result = ACTION_HANDLERS[self.action.action_type]().invoke(session, self.action)
             self.event_logs.create(session=session, status=EventLogStatusChoices.SUCCESS, log=result)
             return result
         except Exception as e:
             logging.error(e)
             self.event_logs.create(session=session, status=EventLogStatusChoices.FAILURE, log=str(e))
 
+    @transaction.atomic()
+    def delete(self, *args, **kwargs):
+        result = super().delete(*args, **kwargs)
+        self.action.delete(*args, **kwargs)
```

Comment:
is this necessary for the `event_action_deleted` to be called?

Reply:
Are you asking if it's needed to delete the `action` in order to call `event_action_deleted`? If so, not really no, but actions are not deleted when its `StaticTrigger` is deleted, which doesn't seem correct, since it has a 1-1 relationship with it. I hope this answers your question?

Reply:
Ah I see. One option would be to move the field so that it's on the action and not the trigger. You could still then use `trigger.action` but deleting a trigger would also delete the action.

Alternatively a `post_delete` signal could be used which also get's fired for queryset deleted.

I'd actually be in favour of changing the relationship since actions can't exist without triggers (but fine to be a separate PR).

Reply:
I just realized that since scheduled messages has an action FK, they will be removed whenever we remove the action. I think this cleanup function was leftover code from the previous iteration. Anycase, see [this](https://github.com/dimagi/open-chat-studio/pull/373/commits/7443a18165d018dd662bad3c7e82e0b814e020d7)

Reply:
OK, but what about the action? Will the action get deleted when deleting the trigger?

Reply:
Yes, since we still keep the delete method on the static trigger that deletes the action